examples/gitlab/gitlab.yml

```diff
@@ -0,0 +1,141 @@
+name: gitlab
+labels:
+    app: gitlab
+replicas: 1
+containers:
+      - name: gitlab
+        image: gitlab/gitlab-ce:9.4.1-ce.0
+        imagePullPolicy: ""
+        env:
+        ## General GitLab Configs
+        ##
+        # This is a free-form env var that GitLab Omnibus uses to configure
+        # everything. We're passing this in from a configmap and pulling some
+        # of the values from the env vars defined below. This is done to
+        # avoid leaving secrets visible in kubectl.
+        - name: GITLAB_OMNIBUS_CONFIG
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab
+              key: gitlab_omnibus_config
+        - name: GITLAB_ROOT_PASSWORD
+        - name: EXTERNAL_URL
+          value: "http://your-domain.com/"
+        ## DB configuration
+        ##
+        - name: DB_HOST
+          value: postgresql
+        - name: DB_USER
+          valueFrom:
+            secretKeyRef:
+              name: gitlab
+              key: db-user
+        - name: DB_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: gitlab
+              key: db-password
+        - name: DB_DATABASE
+          value: "gitlab"
+        ## Redis configuration
+        ##
+        - name: REDIS_HOST
+          value: redis
+        - name: REDIS_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: gitlab
+              key: redis-password
+        ports:
```

Comment:
we can reduce the file size, by removing the `ports` section, from under pods.

so added following changes

```diff
diff --git a/examples/gitlab/gitlab.yml b/examples/gitlab/gitlab.yml
index 4164541..fe798e6 100644
--- a/examples/gitlab/gitlab.yml
+++ b/examples/gitlab/gitlab.yml
@@ -46,17 +46,10 @@ containers:
             secretKeyRef:
               name: gitlab
               key: redis-password
-        ports:
-        - name: ssh
-          containerPort: 22
-        - name: http
-          containerPort: 80
-        - name: https
-          containerPort: 443
         livenessProbe:
           httpGet:
             path: /help
-            port: http
+            port: 80
           # This pod takes a very long time to start up. Be cautious when
           # lowering this value to avoid Pod death during startup.
           initialDelaySeconds: 200
@@ -67,7 +60,7 @@ containers:
         readinessProbe:
           httpGet:
             path: /help
-            port: http
+            port: 80
           initialDelaySeconds: 30
           timeoutSeconds: 1
           periodSeconds: 10
@@ -94,13 +87,13 @@ services:
   ports:
   - name: ssh
     port: 22
-    targetPort: ssh
+    targetPort: 22
   - name: http
     port: 80
-    targetPort: http
+    targetPort: 80
   - name: https
     port: 443
-    targetPort: https
+    targetPort: 443
 
 volumeClaims:
 - name: gitlab-data
diff --git a/examples/gitlab/postgres.yml b/examples/gitlab/postgres.yml
index 7e20906..b4ff9ee 100644
--- a/examples/gitlab/postgres.yml
+++ b/examples/gitlab/postgres.yml
@@ -22,9 +22,6 @@ containers:
               key: postgres-password
         - name: POD_IP
           valueFrom: { fieldRef: { fieldPath: status.podIP } }
-        ports:
-        - name: postgresql
-          containerPort: 5432
         livenessProbe:
           exec:
             command:
@@ -60,7 +57,7 @@ services:
   ports:
   - name: postgresql
     port: 5432
-    targetPort: postgresql
+    targetPort: 5432
 
 volumeClaims:
 - name: data
diff --git a/examples/gitlab/redis.yml b/examples/gitlab/redis.yml
index ac9a87c..4465f15 100644
--- a/examples/gitlab/redis.yml
+++ b/examples/gitlab/redis.yml
@@ -11,9 +11,6 @@ containers:
             secretKeyRef:
               name: redis
               key: redis-password
-        ports:
-        - name: redis
-          containerPort: 6379
         livenessProbe:
           exec:
             command:
@@ -43,7 +40,7 @@ services:
   ports:
   - name: redis
     port: 6379
-    targetPort: redis
+    targetPort: 6379
 
 volumeClaims:
 - name: redis-data

```